eng/scripts/Automation-Sdk-Generate.ps1

```diff
@@ -0,0 +1,137 @@
+[CmdletBinding()]
+param(
+  [Parameter()]
+  [ValidateNotNullOrEmpty()]
+  [string] $InputJsonPath = $GenerateInput,
+
+  [Parameter()]
+  [string] $OutputJsonPath = $GenerateOutput,
+
+  [Parameter()]
+  [string] $RepoRoot = "${PSScriptRoot}/../.."
+)
+
+$inputFileContent = Get-Content $InputJsonPath | ConvertFrom-Json
+$inputFilePaths = $inputFileContent.changedFiles;
+$inputFilePaths += $inputFileContent.relatedReadmeMdFiles;
+$inputFilePaths = $inputFilePaths | Select-Object -Unique
+
+$changedFilePaths = $inputFilePaths -join "`n";
+Write-Host "List of changed swagger files and related readmes:`n$changedFilePaths`n"
+
+$autorestFilesPath = Get-ChildItem -Path "$RepoRoot/sdk"  -Filter autorest.md -Recurse | Resolve-Path -Relative
+
+Write-Host "Updating autorest.md files for all the changed swaggers."
+$sdksInfo = @{}
+$headSha = $inputFileContent.headSha
+$repoHttpsUrl = $inputFileContent.repoHttpsUrl
+foreach ($path in $autorestFilesPath) {
+  foreach ($inputFilePath in $inputFilePaths) {
+    $fileContent = Get-Content $path
+    $isUpdatedLines = $false
+    $escapedInputFilePath = [System.Text.RegularExpressions.Regex]::Escape($inputFilePath)
+    $regexForMatchingShaAndPath = "https:\/\/[^`"]*[\/][0-9a-f]{4,40}[\/]$escapedInputFilePath"
+    $updatedLines = foreach ($line in $fileContent) {
+      if ($line -match $regexForMatchingShaAndPath) {
+        $line -replace $regexForMatchingShaAndPath, "$repoHttpsUrl/blob/$headSha/$inputFilePath"
+
+        $isUpdatedLines = $true
+        $sdkpath = (get-item $path).Directory.Parent.FullName | Resolve-Path -Relative
+        if (!$sdksInfo.ContainsKey($sdkpath)) {
+          $specReadmePath = $inputFileContent.relatedReadmeMdFiles -match $inputFilePath
+          $sdksInfo.Add($sdkpath, $specReadmePath)
+        }
+      }
+      else {
+        $line
+      }
+    }
+    if ($isUpdatedLines) {
+      $updatedLines | Out-File -FilePath $path
+    }
+  }
+}
+
+Write-Host "Updated autorest.md files for all the changed swaggers. `n"
+
+$packages = @()
+foreach ($sdkPath in $sdksInfo.Keys) {
+  $packageName = Split-Path $sdkPath -Leaf
+  Write-Host "Generating code for " $packageName
+  $artifacts = @()
+  $hasBreakingChange = $null
+  $content = $null
+  $srcPath = Join-Path $sdkPath 'src'
+  dotnet msbuild /restore /t:GenerateCode $srcPath
+  if (!$LASTEXITCODE) {
+    $result = "succeeded"
+    Write-Host "Successfully generated code for" $packageName "`n"
+    
+    $csprojPath = Get-ChildItem $srcPath -Filter *.csproj -Recurse
+    dotnet pack $csprojPath /p:RunApiCompat=$false
```

Comment:
Sometimes when there are breaking changes the build might fail so you would never get to `dotnet build` line. Not sure how we want to handle that.

Reply:
If the build will fail it go to the else block [here](https://github.com/Azure/azure-sdk-for-net/blob/6913ed4d48856a206f98d2b728750a3e9ffe2673/eng/scripts/Automation-Sdk-Generate.ps1#L91) and It will still create a PR. Then we will have manually fix the issues.

Check Digital twin example [here](https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=4264729&view=logs&j=7f026a02-f755-5ce4-425c-2d5df22c9d55&t=20b436ef-0412-5b55-7996-5294882c57b5&l=193), it failed while creating an artifacts but [here ](https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=4264729&view=logs&j=7f026a02-f755-5ce4-425c-2d5df22c9d55&t=20b436ef-0412-5b55-7996-5294882c57b5&l=341)it created a PR

Reply:
Depending on the changes it the build might fail but it should be mostly limited to the individual project and not span to other projects because we don't use project references.